documentation/specs/TextField.md

```diff
@@ -0,0 +1,307 @@
+          <Text>{label}</Text>
+        </label>
+      </div>
+      {showHelperText && (
+        <div {...helperTextProps}>
+          <Text>{helperText}</Text>
+        </div>
+      )}
+      {showErrorText && (
+        <div {...errorTextProps}>
+          <Text>{errorText}</Text>
+        </div>
+      )}
+    </div>
+  );
+}
+```
+
+### Example Usage
+
+_Email with autoFocus:_
+
+```tsx
+import { TextField } from "@easypost/easy-ui/TextField";
+
+export function Component() {
+  const [email, setEmail] = useState("");
+  return (
+    <>
+      <TextField
+        type="email"
+        label="Email address"
+        autoFocus
+        value={email}
+        onChange={(inputValue) => setEmail(inputValue)} // value is returned automatically via react-aria
+        isRequired
+      />
+      <span>email entered: {email}</span>
+    </>
+  );
+}
+```
+
+_Protected variation with helper text:_
+
+```tsx
+import { TextField } from "@easypost/easy-ui/TextField";
+
+export function Component() {
+  const [password, setPassword] = useState("");
+  return (
+    <>
+      <TextField
+        variant="protected"
+        label="Password"
+        helperText={<a href={RESET_URL}>Forgot password? </a>}
+        value={password}
+        onChange={(inputValue) => setPassword(inputValue)} // value is returned automatically via react-aria
+        isRequired
+      />
+    </>
+  );
+}
+```
+
+_Visually hidden label with left aligned icon and placeholder text:_
+
+```tsx
+import { TextField } from "@easypost/easy-ui/TextField";
+import AnIcon from "@easypost/easy-ui-icons/AnIcon";
+
+export function Component() {
+  const [searchedValue, setSearchedValue] = useState("");
+  return (
+    <>
+      <TextField
+        type="search"
+        label="Search for carriers" // visually hidden but still accessible via isLabelVisuallyHidden prop
+        isLabelVisuallyHidden
+        leftAlignedIcon={AnIcon}
+        value={searchedValue}
+        onChange={(inputValue) => setSearchedValue(inputValue)} // value is returned automatically via react-aria
+        placeholder="FedEx, UPS, USPS"
+        isRequired
+      />
+    </>
+  );
+}
+```
+
+_Disabled TextField:_
+
+```tsx
+import { TextField } from "@easypost/easy-ui/TextField";
+
+export function Component() {
+  const [value, setValue] = useState("");
+  return (
+    <>
+      <TextField
+        label="Disabled"
+        value={value}
+        onChange={(inputValue) => setValue(inputValue)} // value is returned automatically via react-aria
+        isDisabled
+      />
+    </>
+  );
+}
+```
+
+_Invalid state with error text:_
+
+```tsx
+import { TextField } from "@easypost/easy-ui/TextField";
+
+export function Component() {
+  const [value, setValue] = useState("");
+  return (
+    <>
+      <TextField
+        label="Error"
+        validationState="invalid"
+        helperText="Some text" // will be overriden in the presence of an invalid state with error text
+        errorText="Some error text"
```

Comment:
**thought:** having both a `helperText` and `errorText` for the same positional slot tied to different states of the component feels just slightly odd to me, however it probably makes the most sense to do it this way, otherwise the consumer has to think about updating the slot based on validation logic